common/djangoapps/third_party_auth/lti.py

```diff
@@ -0,0 +1,194 @@
+"""
+Third-party-auth module for Learning Tools Interoperability
+"""
+import logging
+import calendar
+import time
+
+from django.contrib.auth import REDIRECT_FIELD_NAME, logout
+from oauthlib.common import Request
+from oauthlib.oauth1.rfc5849.signature import (
+    normalize_base_string_uri,
+    normalize_parameters,
+    collect_parameters,
+    construct_base_string,
+    sign_hmac_sha1,
+)
+from social.backends.base import BaseAuth
+from social.exceptions import AuthFailed
+from social.utils import sanitize_redirect
+
+log = logging.getLogger(__name__)
+
+LTI_PARAMS_KEY = 'tpa-lti-params'
+
+
+class LTIAuthBackend(BaseAuth):
+    """
+    Third-party-auth module for Learning Tools Interoperability
+    """
+
+    name = 'lti'
+
+    def start(self):
+        """
+        Prepare to handle a login request. This method replaces
+        social.actions.do_auth and must be kept in sync with any
+        upstream changes in that method.
+        """
+
+        # Clean any partial pipeline data
+        self.strategy.clean_partial_pipeline()
+
+        # Save validated LTI parameters (or None if invalid or not submitted)
+        validated_lti_params = self.get_validated_lti_params(self.strategy)
+        if not validated_lti_params:
+            self.strategy.session_set(LTI_PARAMS_KEY, None)
+            raise AuthFailed(self)
+        else:
+            logout(self.strategy.request)
+            self.strategy.session_set(LTI_PARAMS_KEY, validated_lti_params)
+
+        # Save extra data into session.
+        # While Basic LTI 1.0 specifies that the message is to be signed using OAuth, implying
+        # that any GET parameters should be stripped from the base URL and included as signed
+        # parameters, typical LTI Tool Consumer implementations do not support this behaviour. As
+        # a workaround, we accept TPA parameters from LTI custom parameters prefixed with "tpa_".
+
+        for field_name in self.setting('FIELDS_STORED_IN_SESSION', []):
+            if 'custom_tpa_' + field_name in validated_lti_params:
+                self.strategy.session_set(field_name, validated_lti_params['custom_tpa_' + field_name])
+
+        if 'custom_tpa_' + REDIRECT_FIELD_NAME in validated_lti_params:
+            # Check and sanitize a user-defined GET/POST next field value
+            redirect_uri = validated_lti_params['custom_tpa_' + REDIRECT_FIELD_NAME]
+            if self.setting('SANITIZE_REDIRECTS', True):
+                redirect_uri = sanitize_redirect(self.strategy.request_host(), redirect_uri)
+            self.strategy.session_set(REDIRECT_FIELD_NAME, redirect_uri or self.setting('LOGIN_REDIRECT_URL'))
+
+        # Set a auth_entry here so we don't have to receive that as a custom parameter
+        self.strategy.session_setdefault('auth_entry', 'login')
+
+    def auth_html(self):
+        """
+        Not used
+        """
+        pass
```

Comment:
Nit: I assume this is only here to suppress a pylint warning. I think you should use `raise NotImplementedError` instead of `pass`, since either way will address the pylint issue but the exception would provide useful feedback to anyone who accidentally calls this method.